src/main/java/org/broadinstitute/hellbender/utils/iterators/IntervalAlignmentContextIterator.java

```diff
@@ -23,7 +24,15 @@
     private AlignmentContext currentAlignmentContext;
     private SAMSequenceDictionary dictionary;
 
-    public IntervalAlignmentContextIterator(final Iterator<AlignmentContext> alignmentContextIterator, final IntervalLocusIterator intervalLocusIterator, final SAMSequenceDictionary dictionary) {
+
+    /**
+     *  Note:  Typically, if you are calling this from a walker tool, you want to use {@link AlignmentContextIteratorFactory}
+     *
+     * @param alignmentContextIterator iterator for alignment context
+     * @param intervalLocusIterator an iterator that will traverse all relevant loci
+     * @param dictionary reference or best available dictionary.
+     */
+    public IntervalAlignmentContextIterator(Iterator<AlignmentContext> alignmentContextIterator, IntervalLocusIterator intervalLocusIterator, SAMSequenceDictionary dictionary) {
```

Comment:
I will make this live in the `org.broadinstitute.hellbender.utils.locusiterator` with the factory, and make package-protected. Reduce the visibility will avoid incorrect usages and you can remove the note.

It could also live as an static class inside the factory, but I always prefer not to implement nested classes...